mycroft/skills/core.py

```diff
@@ -691,17 +703,59 @@ def update_event(self, name, data=None):
         }
         self.emitter.emit(Message('mycroft.schedule.update_event', data=data))
 
-    def cancel_event(self, name):
+    def cancel_scheduled__event(self, name):
         """
             Cancel a pending event. The event will no longer be scheduled
             to be executed
 
             Args:
                 name (str):   Name of event
         """
-        data = {'event': self._unique_name(name)}
+        unique_name = self._unique_name(name)
+        data = {'event': unique_name}
+        self.remove_event(unique_name)
         self.emitter.emit(Message('mycroft.scheduler.remove_event', data=data))
 
+    def get_scheduled_event_status(self, name):
+        """
+            Get scheduled event data and return the amount of time left
+
+            Args:
+                name (str): Name of event
+
+            Return:
+                time_left (int): the time left in seconds
+        """
+        event_name = self._unique_name(name)
+        data = {'name': event_name}
+
+        # making event_status an object so it's refrence can be changed
+        event_status = [None]
+
+        def callback(message, event_status):
+            LOG.info("skill-timer {}".format(message.data))
+            if message.data is None:
+                event_status[0] = False
+            else:
+                event_time = int(message.data[0][0])
+                current_time = int(time.time())
+                time_left_in_seconds = event_time - current_time
+                event_status[0] = time_left_in_seconds
+
+        emitter_name = 'mycroft.event_status.callback.{}'.format(event_name)
+        self.emitter.once(emitter_name,
+                          lambda message: callback(message, event_status))
+        self.emitter.emit(Message('mycroft.scheduler.get_event', data=data))
+
+        start_wait = time.time()
+        while event_status[0] is None and time.time() - start_wait < 3.0:
+            time.sleep(0.1)
+        if event_status[0] is False:
+            raise Exception("Scheduled Event not found for {}".format(name))
```

Comment:
Sorry if I was unclear (or confused, or both) earlier. I meant that the function should return `None` if there is no scheduled event.

A use case I can imagine is to check if an event is scheduled at all and checking if The `self.get_scheduled_event_status(event) is None` is a pretty simple way to do it.

Timeout is an actual error while "event isn't scheduled" is a valid status (at least in my mind).

What's your opinion?

Reply:
makes a lot of sense to me!